src/renwindow.c

```diff
@@ -33,11 +34,11 @@ void renwin_init_surface(RenWindow *ren) {
   ren->scale_x = ren->scale_y = 1;
 #ifdef LITE_USE_SDL_RENDERER
   if (ren->rensurface.surface) {
-    SDL_FreeSurface(ren->rensurface.surface);
+    SDL_DestroySurface(ren->rensurface.surface);
   }
   int w, h;
-  SDL_GL_GetDrawableSize(ren->window, &w, &h);
-  ren->rensurface.surface = SDL_CreateRGBSurfaceWithFormat(0, w, h, 32, SDL_PIXELFORMAT_BGRA32);
+  SDL_GetWindowSizeInPixels(ren->window, &w, &h);
+  ren->rensurface.surface = SDL_CreateSurface(w, h, SDL_PIXELFORMAT_BGRA32);
```

Comment:
I wonder why this format was chosen. Is there a way to know the optimal format for a system?

Should this format match the `draw_rect_surface` one from `renderer.c`?

Reply:
The RENDERER backend now uses the Window surface format.
For `draw_rect_surface`, most systems returns a surface format that has no alpha (like XRGB8888), so we simply can't tie it to the system surface format.